liota/dccs/iotcc.py

```diff
@@ -335,10 +376,26 @@ def set_properties(self, reg_entity_obj, properties):
         else:
             entity = reg_entity_obj.ref_entity
 
-        log.info("Properties defined for resource {0}".format(entity.name))
+        def on_response(msg):
+            try:
+                log.debug("Received msg: {0}".format(msg))
+                json_msg = json.loads(msg)
+                log.debug("Processing msg: {0}".format(json_msg["type"]))
+                self._check_version(json_msg)
+                if json_msg["type"] == "add_properties_response" and json_msg["body"][
```

Comment:
We should have transaction id in all these calls, and validate based on transaction id. In case of out-of-order processing and repetition of messages, responses can get jumbled.

Reply:
In the updated PR, transaction_id is now used for validating in "_dispatch_recvd_msg".

Reply:
The on_response() message is different for each request, in case of registration it returns uuid and in case of get_properties it is different as it returns the property list.  Also, there is specific check in the on_response call to process only the message which meets its entities criteria. Hence, better to have them separate as of now. Agreed in v2 we can surely refactor to have them common.

Reply:
For the existing on_response() methods in register and get_properties, the actual response gets processed to extract information. However, now we are introducing too many on_response methods  which can be handled by common code. Hence, it might be better to introduce new code with a common method, rather than duplicating it

Reply:
So we keep separate implementation for the request which needs to extract information on-response and separate refactoring for the call which just needs a successful check. Let`s discuss it on the call.

Reply:
As discussed have refactored the code and have extracted out the processing of response messages for create_relationship, set_properties, set_system_properties and unregister requests. 

@pmasrani  Review the final code changes after incorporating all the review comments.